_posts/2024-04-12-learn-from-the-xz-backdoor.md

```diff
@@ -0,0 +1,111 @@
+
+However, relying on heroes is not a sustainable and reliable strategy. So for the future, we all need to learn from what happened and need to become a large team of small heroes.
+
+
+## Time to look forward and learn from the mistakes
+
+Many words have been written on what happened. Timelines have been written, actors named. The backdoor has been reversed and the understanding of the purpose of the code has been published in a lot of detail.
+
+It's time to look forward. Linux distributions were abused to deliver a backdoor to their users. What the exact purpose of the backdoor was is still speculation. It could be all from an individual who wanted to sell access to abundant compute power via public cloud hosted virtual machines that have a vulnerable ssh port open to the public. Which is the rather unlikely, but still possible, one end of the spectrum. The other end of the spectrum is a company that sells backdoors to state actors that make use of those to remotely and covertly access any linux machine. Although mistakes were made, it almost achieved that goal. Where is the truth? Further evidence needs to be identified and analyzed for that.
+
+
+### Linus Law and the distributions
+
+"*Given enough eyeballs, all bugs are shallow*". In Open Source communities, this is cited a lot as a reason for why Open Source can be trusted. For Open Source projects that are attracting enough attention from sufficiently skilled contributors this "law" probably has at least some weight. However, we learned for example from Heartbleed that these preconditions are not universally fulfilled. There are are many projects that are absolutely essential and yet are considered boring and fail to attract a lot of maintainers or contributors, and those who are on the project are buried under a pile of work already and can't really spend significant effort on ramping up new joiners.
+
+However, as explained earlier, this backdoor was only targetting distributions. First, by the prechecks that the backdoor executed before unfolding, but also because the conditions necessary for implanting were only existing *downstream* in these distributions. Debian, as well as the other affected distributions like openSUSE are carrying significant amount of downstream only patches to essential open source projects, like in this case openssh. With hindsight, that should be another heartbleed level learning for the work of the distributions. These patches were building the essential steps to embed the backdoor, and have not the scrutiny that they likely would have received by the respective upstream maintainers. Whether you trust Linus Law or not, it was not even given a chance to chime in here. Upstream did not fail on the users, distributions failed on upstream and their users here.
```

Comment:
```suggestion
However, as explained earlier, this backdoor was only targetting distributions. First, by the checks that the backdoor executed before unfolding, but also because the conditions necessary for implanting only existed *downstream* in these distributions. Debian, as well as the other affected distributions like openSUSE, carry significant amount of downstream-only patches to essential open-source projects, like in this case openssh. With hindsight, that should be another Heartbleed-level learning for the work of the distributions. These patches were building the essential steps to embed the backdoor, and have not the scrutiny that they likely would have received by the respective upstream maintainers. Whether you trust Linus' Law or not, it was not even given a chance to chime in here. Upstream did not fail on the users, distributions failed on upstream and their users here.
```

Reply:
the conditions still exist, they were not removed, so keep and use present tense i would say. (e.g. presence of debian/rules file, and RPM_ARCH=x86_64 )